File: rhBackEnd/src/main/java/com/rh/entities/Conge.java
package com.rh.entities;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "conge")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Conge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idConge;

    private int nbJoursOuvrables;

    private int nbJoursTotal;

    private int status ;

    private String raison;

    private String typeConge;

    private Date dateDebut;

    private Date dateFin;
    @CreationTimestamp
    private Date dateDemande;

    public Long getIdConge() {
        return idConge;
    }

    public void setIdConge(Long idConge) {
        this.idConge = idConge;
    }

    @ManyToOne
    @JoinColumn(name = "fk_employee")
    Employee employee;

    @ManyToOne
    @JoinColumn(name = "fk_response_conge")
    ResponseConge responseConge;


}

File: rhFront/src/app/employees/Entities/conge.ts
import {Employee} from './employee';
import {ResponseConge} from './response-conge';

export class Conge {
  public idConge: number;
  public nbJoursOuvrables: number;
  private _nbJoursTotal:number = 30;
  public status: number;
  // tslint:disable-next-line:ban-types
  public raison: string;
  public typeConge: string;
  public dateDebut: Date;
  public dateFin: Date;
  public dateDemande: Date;
  public employee1: Employee;
  public employee: string;
  public responseConge: ResponseConge;


  get nbJoursTotal(): number {
    return this._nbJoursTotal;
  }

  set nbJoursTotal(value: number) {
    this._nbJoursTotal = value;
  }
}
